src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -59,7 +62,17 @@ public ConfidentialClientExecutor(IServiceBundle serviceBundle, ConfidentialClie
                 _confidentialClientApplication.AppTokenCacheInternal).ConfigureAwait(false);
        
             requestParams.SendX5C = clientParameters.SendX5C ?? false;
-            
+
+            if (ServiceBundle.Config.IsManagedIdentity)
+            {
+                ManagedIdentityClient managedIdentityClient = new ManagedIdentityClient(requestContext);
+                ServiceBundle.Config.AppTokenProvider = managedIdentityClient.AppTokenProviderImplAsync;
+                // TODO: disable instance discovery
```

Comment:
There should be, not sure if it is [this](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3242) @bgavrilMS can confirm this.

Reply:
@trwalke is picking this up as part of a different feature. But there will still be a flag to set, i.e. replace the custom metadata string with a "disable_metadata=true" option. This should be tracked by the MSI logic, but blocked for now.

Reply:
- I think the metadata related stuff needs to be handled in one of the instance discovery classes. I think we add the known metadata endpoints by default anyway? 
- **Is managed identity enabled per request or per CCA instance?** As-is, since it's only for client creds request, we don't want to disable instance discovery completely, we just want to make sure the known metadata is there, which is needed for this request.
- With the current code, then someone can use client credentials flow with Managed Identity and then a user flow without MI. So for a sovereign cloud they can call WithInstanceDiscoveryMetadata. So setting `CustomInstanceDiscoveryMetadata` below, will actually end up overwriting the user specified one.

Reply:
@pmaytak - agreed, instance discovery should be handled elsewhere, not here. You are right that we don't want to disable instance discovery for the entire CCA, this is a great point. 

@neha-bhargava - maybe it's easier to add a check in the InstanceDiscoevryManager, something like "if MSI is enabled, use only hardcoded metadata or default metadata" ?